PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/test/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/ServiceBusReceiverAsyncClientCheckTest.java

```diff
@@ -0,0 +1,125 @@
+package com.microsoft.azure.toolkit.intellij.azure.sdk.buildtool;
+
+
+// Import necessary libraries
+import com.intellij.ide.highlighter.JavaFileType;
+import com.intellij.openapi.fileTypes.FileType;
+import com.intellij.psi.JavaElementVisitor;
+import com.intellij.psi.PsiElementVisitor;
+import com.intellij.psi.PsiFile;
+import com.intellij.psi.PsiFileFactory;
+import com.intellij.psi.PsiTypeElement;
+import com.intellij.codeInspection.ProblemsHolder;
+import com.intellij.psi.PsiType;
+import com.intellij.openapi.project.Project;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+
+
+public class ServiceBusReceiverAsyncClientCheckTest {
+
+    // Create a mock ProblemsHolder to be used in the test
+    private ProblemsHolder mockHolder;
+
+    // Create a Java code snippet that uses ServiceBusReceiverAsyncClient with situations that should be flagged
+    private String createJavaCode() {
+        StringBuilder javaCode = new StringBuilder();
+
+        javaCode.append("public class TestClass {\n");
+
+        // Define the client as a private final member
+        javaCode.append("\tprivate final ServiceBusReceiverAsyncClient client;\n");
+
+        // Create a constructor for TestClass
+        javaCode.append("\tpublic TestClass() {\n");
+        javaCode.append("\t\tthis.client = new ServiceBusReceiverAsyncClient();\n");
+        javaCode.append("\t}\n");
+
+        // Create a getClient method to access the client
+        javaCode.append("\tpublic ServiceBusReceiverAsyncClient getClient() {\n");
+        javaCode.append("\t\treturn this.client;\n");
+        javaCode.append("\t}\n");
+
+        javaCode.append("}");
+
+        return javaCode.toString();
+    }
+
+
+    @Test
+    public void testBuildVisitor() {
+        // Arrange
+        String javaCode = createJavaCode();
+        PsiElementVisitor visitor = createVisitor();
+
+        // Mock PsiFileFactory
+        PsiFileFactory psiFileFactory = mockPsiFileFactory();
+
+        // Create PsiFile
+        PsiFile psiFile = createPsiFile(javaCode, psiFileFactory);
+
+        // Accept visitor
+        psiFile.accept(visitor);
+
+        // Assert
+        assertVisitor(visitor);
+
+        // Visit Type Element
+        visitTypeElement(visitor);
+    }
+
```

Comment:
Lets add one more test for [negative/unhappy path testing](https://www.tutorialspoint.com/what-is-negative-testing-test-cases-with-examplehttps://www.tutorialspoint.com/what-is-negative-testing-test-cases-with-example).
>Many developers make the mistake of only testing happy paths, or what users are supposed to do when using an application, and neglect testing unhappy paths, or the many ways that users can break your software. By doing so, they may not be prepared to properly handle errors.

Reply:
I added a test to check when PsiTypeElement is null. I also wanted to add a test to check instances that registerProblem receives the wrong type of object, but the visitor never visits the wrong type of element so the test wouldn't run

Reply:
add one more test case later

Reply:
extra test added!